src/display/api.js

```diff
@@ -3089,7 +3089,9 @@ class WorkerTransport {
       this.fontLoader.clear();
     }
     this.#methodPromises.clear();
-    this.filterFactory.destroy();
+    if (this.filterFactory.isCleanable()) {
+      this.filterFactory.destroy();
+    }
```

Comment:
I'd prefer if we handle this with an optional parameter instead:
```suggestion
    this.filterFactory.destroy(/* keepHcm = */ true);
```

And thus also change:
 - https://github.com/mozilla/pdf.js/blob/5e4b3d13ebc6eb2453f0cad12f33964e2d4fc6fc/src/display/display_utils.js#L175-L176 into
   ```javascript
   destroy(keepHcm = false) {
     if (keepHcm && this.#hcmUrl) {
       return;
     }
     if (this.#_defs) {
       ...
   ```

 - https://github.com/mozilla/pdf.js/blob/5e4b3d13ebc6eb2453f0cad12f33964e2d4fc6fc/src/display/base_factory.js#L29 into
   ```javascript
   destroy(keepHcm = false) {}
   ```